controllers/flowlogspipeline/flp_objects.go

```diff
@@ -33,16 +33,55 @@ const (
 
 // PodConfigurationDigest is an annotation name to facilitate pod restart after
 // any external configuration change
-const PodConfigurationDigest = "flows.netobserv.io/goflow-kube-config"
+const PodConfigurationDigest = "flows.netobserv.io/" + configMapName
 
-type ConfigMap struct {
-	Listen      string        `json:"listen,omitempty"`
-	Loki        LokiConfigMap `json:"loki,omitempty"`
-	PrintInput  bool          `json:"printInput"`
-	PrintOutput bool          `json:"printOutput"`
+type Config struct {
+	LogLevel string   `json:"log-level,omitempty"`
+	Pipeline Pipeline `json:"pipeline,omitempty"`
+	Health   Health   `json:"health,omitempty"`
 }
 
-type LokiConfigMap struct {
+type Health struct {
+	Port int `json:"port"`
+}
+
+type Pipeline struct {
+	Ingest    Ingest      `json:"ingest"`
+	Decode    Decode      `json:"decode"`
+	Encode    Encode      `json:"encode"`
+	Extract   Extract     `json:"extract"`
+	Transform []Transform `json:"transform"`
+	Write     Write       `json:"write"`
+}
```

Comment:
This structure with change after this PR: https://github.com/netobserv/flowlogs-pipeline/pull/83

Anyway, I think this structure will remain constant. Am I right? In that case, for simplification, we can just bundle the configuration as a YAML text/file and embed it into the FLP Pod descriptor.

Reply:
It's not entirely constant, in particular the Loki part is exposed through CR